### PR TITLE
Vertex shaders & vertex loader -> x64 JIT 

### PR DESCRIPTION
**Pull request by wheremyfoodat**
Supersedes #15

### CODE REVIEW

File: src/core/PICA/dynapica/shader_rec_emitter_x64.cpp [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+
+void ShaderEmitter::storeRegister(Xmm source, const PICAShader& shader, u32 dest, u32 operandDescriptor) {
+	const vec4f& destRef = getDestRef(shader, dest);
+	const uintptr_t offset = uintptr_t(&destRef) - uintptr_t(&shader); // Calculate offset of register from start of the state struct
+
+	// Mask of which lanes to write
+	u32 writeMask = operandDescriptor & 0xf;
+	if (writeMask == 0xf) { // No lanes are masked, just movaps
+		movaps(xword[statePointer + offset], source);
+	} else if (std::popcount(writeMask) == 1) { // Only 1 register needs to be written back. This can be done with a simple shift right + movss
```
Comment (Wunkolo) [inline src/core/PICA/dynapica/shader_rec_emitter_x64.cpp:289]: [std::has_single_bit](https://en.cppreference.com/w/cpp/numeric/has_single_bit)

File: src/core/PICA/dynapica/shader_rec_emitter_x64.cpp [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+void ShaderEmitter::recMAX(const PICAShader& shader, u32 instruction) {
+	const u32 operandDescriptor = shader.operandDescriptors[instruction & 0x7f];
+	const u32 src1 = getBits<12, 7>(instruction);
+	const u32 src2 = getBits<7, 5>(instruction); // src2 coming first because PICA moment
+	const u32 idx = getBits<19, 2>(instruction);
+	const u32 dest = getBits<21, 5>(instruction);
+
+	loadRegister<1>(src1_xmm, shader, src1, idx, operandDescriptor);
+	loadRegister<2>(src2_xmm, shader, src2, 0, operandDescriptor);
+	maxps(src1_xmm, src2_xmm);
```
Comment (Wunkolo) [inline src/core/PICA/dynapica/shader_rec_emitter_x64.cpp:486]: Should be worth noting somewhere here(prob as a `TODO`) that [PICA handles min/max different from x86](https://www.3dbrew.org/wiki/GPU/Shader_Instruction_Set#Floating-Point_Behavior). Same for `rsq` and `rcp`

Comment (wheremyfoodat) [reply src/core/PICA/dynapica/shader_rec_emitter_x64.cpp:486]: I am not quite sure it's possible to get rsq/rcp to ever have the same precision without f24 soft float. Min/max are easier yeah, though thankfully games don't seem to think about them too much.

**Comment by Wunkolo** [comment]
panda 🎉

**Comment by wheremyfoodat** [comment]
panda !

**Comment by fleroviux** [comment]
panda!

**Pull request merged**